### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by StasOkhrym**

### CODE REVIEW

**Review by mltdll — APPROVED** [review]

File: db/models.py [hunk 1]
```diff
@@ -14,3 +14,35 @@ class Actor(models.Model):
 
     def __str__(self) -> str:
         return f"{self.first_name} {self.last_name}"
+
+
+class Movie(models.Model):
+    title = models.CharField(max_length=255)
+    description = models.TextField()
+
+    actors = models.ManyToManyField(Actor, related_name="actors")
```
Comment (Y-Havryliv) [inline db/models.py:23]: `related_name` must be movies

File: db/models.py [hunk 2]
```diff
@@ -14,3 +14,35 @@ class Actor(models.Model):
...
+    genres = models.ManyToManyField(Genre, related_name="genres")
```
Comment (Y-Havryliv) [inline db/models.py:24]: same here

File: db/models.py [hunk 3]
```diff
@@ -14,3 +14,35 @@ class Actor(models.Model):
...
+
+    def __str__(self) -> str:
+        return f"{self.title}"
```
Comment (Y-Havryliv) [inline db/models.py:27]: do you actually need f-string here?

File: db/models.py [hunk 4]
```diff
@@ -14,3 +14,35 @@ class Actor(models.Model):
...
+class CinemaHall(models.Model):
+    name = models.CharField(max_length=255)
+    rows = models.IntegerField()
+    seats_in_row = models.IntegerField()
+
+    def capacity(self) -> int:
+        return self.rows * self.seats_in_row
+
+    def __str__(self) -> str:
+        return f"{self.name}"
```
Comment (Y-Havryliv) [inline db/models.py:39]: do you actually need f-string here?

File: services/cinema_hall.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+def get_cinema_halls() -> QuerySet:
+    return CinemaHall.objects.all()
+
+
+def create_cinema_hall(
+        hall_name: str,
+        hall_rows: int,
+        hall_seats_in_row: int
+) -> None:
+    CinemaHall.objects.create(
```
Comment (Y-Havryliv) [inline services/cinema_hall.py:14]: it is a good practice when your functions return smth

**Review by Y-Havryliv — CHANGES_REQUESTED** [review]

File: services/cinema_hall.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+def get_cinema_halls() -> QuerySet:
+    return CinemaHall.objects.all()
+
+
+def create_cinema_hall(
+        hall_name: str,
+        hall_rows: int,
+        hall_seats_in_row: int
+) -> QuerySet:
```
Comment (Y-Havryliv) [inline services/cinema_hall.py:13]: it doesn't return `QuerySet`

File: services/movie_session.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
+from django.db.models import QuerySet, DateTimeField
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: DateTimeField,
+        movie_id: int,
+        cinema_hall_id: int
+) -> QuerySet:
```
Comment (Y-Havryliv) [inline services/movie_session.py:9]: same here

**Review by Y-Havryliv — APPROVED** [review]
Nice!

**Pull request closed without merging**